saleor/graphql/meta/mutations.py

```diff
@@ -0,0 +1,185 @@
+import graphene
+from django.core.exceptions import ValidationError
+from graphql_jwt.exceptions import PermissionDenied
+
+from ...core import models
+from ...core.error_codes import MetadataErrorCode
+from ..core.mutations import BaseMutation
+from ..core.types.common import MetadataError
+from .permissions import PRIVATE_META_PERMISSION_MAP, PUBLIC_META_PERMISSION_MAP
+from .types import ObjectWithMetadata
+
+
+class MetadataPermissionOptions(graphene.types.mutation.MutationOptions):
+    permission_map = {}
+
+
+class BaseMetadataMutation(BaseMutation):
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def __init_subclass_with_meta__(
+        cls, arguments=None, permission_map=[], _meta=None, **kwargs,
+    ):
+        if not _meta:
+            _meta = MetadataPermissionOptions(cls)
+        if not arguments:
+            arguments = {}
+        fields = {"item": graphene.Field(ObjectWithMetadata)}
+
+        _meta.permission_map = permission_map
+
+        super().__init_subclass_with_meta__(_meta=_meta, **kwargs)
+        cls._update_mutation_arguments_and_fields(arguments=arguments, fields=fields)
+
+    @classmethod
+    def get_instance(cls, info, **data):
+        object_id = data.get("id")
+        if object_id:
+            try:
+                instance = cls.get_node_or_error(info, object_id)
+            except ValidationError:
+                instance = None
+            if instance and issubclass(type(instance), models.ModelWithMetadata):
+                return instance
+        raise ValidationError(
+            {
+                "id": ValidationError(
+                    f"Couldn't resolve to a item with meta: {object_id}",
+                    code=MetadataErrorCode.NOT_FOUND.value,
+                )
+            }
+        )
+
+    @classmethod
+    def get_permissions(cls, info, **data):
+        object_id = data.get("id")
+        if not object_id:
+            return []
+        type_name, object_pk = graphene.Node.from_global_id(object_id)
+        permission = cls._meta.permission_map.get(type_name)
+        if permission:
+            return permission(info, object_pk)
+        raise ValidationError(
```

Comment:
I don't understand in what cases this validation would be raised? How is it possible that we fail to resolve permission for an item? We control the available metadata objects and should make sure that we provide permissions for them.

Reply:
Should raise `NotImplementedError`